Lib/subprocess.py

```diff
@@ -919,6 +955,32 @@ def _check_timeout(self, endtime, orig_timeout):
             raise TimeoutExpired(self.args, orig_timeout)
 
 
+    def wait(self, timeout=None):
+        """Wait for child process to terminate; returns self.returncode."""
+        if timeout is not None:
+            endtime = _time() + timeout
+        try:
+            return self._wait(timeout=timeout)
+        except KeyboardInterrupt:
+            # https://bugs.python.org/issue25942
+            # The first keyboard interrupt waits briefly for the child to
+            # exit under the common assumption that it also received the ^C
+            # generated SIGINT and will exit rapidly.  A second ^C will
+            # abort this immediately.  Matching user patterns of hitting
+            # ^C more than once before resorting to ^\ (SIGKILL).
```

Comment:
This is not very clear to me. You should mention where the SIGKILL is emitted.

Reply:
A second ^C will interrupt this exception handler and not be caught by anything.

SIGKILL comes from the user hitting ^\ when they are impatient at ^C not having terminated the command line tool they were trying to exit.

make sense?   I could just get rid of everything after the first sentence.

Reply:
Uh, I see. The "resorting to SIGKILL" sounded like it was done by `subprocess` itself (especially as `p.kill` is called elsewhere).